src/metabase/lib/binning.cljc

```diff
@@ -194,3 +203,32 @@
           {:bin-width bin-width
            :min-value value
            :max-value (+ value bin-width)})))))
+
+(defn- binning_info->binning-options
+  [binning_info]
+  (-> binning_info
+      u/kebab-keys
+      (set/rename-keys {:binning-strategy :strategy})))
+
+(defn ends-with-binning?
+  "Decide whether string `s` has binning suffix based on `binning-options`."
+  [s binning-options semantic-type]
+  (str/ends-with? s (str ": " (binning-display-name binning-options semantic-type))))
+
+(defn ensure-ends-with-binning
+  "Ensure that `s` is suffixed by appropriate binning info."
+  [s binning-options semantic-type]
+  (if (or (not (string? s))
+          (not (:strategy binning-options))
+          (ends-with-binning? s binning-options semantic-type))
+    s
+    (lib.util/format "%s: %s" s (binning-display-name binning-options semantic-type))))
```

Comment:
This is the kind of "guesswork" that makes me uncomfortable. When does it get duplicated? It feels likely we can find a better way to avoid duplicate suffixes than examining the string.

Reply:
Call to `ensure-binning-in-display-name`, the next function, was added into `models/interface.clj` and `middleware/annotate.clj`. To the former, to ensure that loaded cards, saved prior this change is merged, have binning in display name, and to the latter so the cards' columns are saved with display name that reflects column's binning.

Hence, without this check, loading a card would result in duplicating the binning in display name, for cards that have the binning in display name already.

Alternative solution would require adding another key to result_metadata, so it could be decided whether the loaded card was saved prior to this change. The new `:was_binned` key _could_ be used for that. 

Bu then, this function, `ensure-ends-with-binning`, is used also in `lib.metadata.calculation/display-name-method :metadata/column` in `field.clj`, thanks to this check we avoid rendering duplicate binning in display names for columns that were binning by same binning on eg. 2 following stages. (Doing that does not make much sense, but prior to this change if I wanted to breakout by column binned on previous stage qb would provide me with default binning and I had to manually change that to don't bin. Lot of e2e tests were written in a way that double binning like this would occur.)

Reason why I decided for going down this road was simplicity -- no need to add anything besides `was_binned` to results metadata. I think it would be great to have this solved without string matching, but as per our recent discussion, I also try to minimize changes to mbql (ie. nothing more fancy as bool in this PR; solving display name would require probably also the binning of source column) while field refs work is in progress. 

I believe it could be a good idea to tackle this with field refs overhaul merged. WDYT?